src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,287 @@
+        /// <param name="ak8963Interface">The specific interface to communicate with the AK8963</param>
+        /// <param name="autoDispose">True to dispose the I2C device when class is disposed</param>
+        public Ak8963(I2cDevice i2CDevice, Ak8963I2cBase ak8963Interface, bool autoDispose = true)
+        {
+            _i2cDevice = i2CDevice;
+            _ak8963Interface = ak8963Interface;
+            // Initialize the default modes
+            _measurementMode = MeasurementMode.PowerDown;
+            _outputBitMode = OutputBitMode.Output14bit;
+            _autodispose = autoDispose;
+            byte mode = (byte)((byte)_measurementMode | ((byte)_outputBitMode << 4));
+            WriteRegister(Register.CNTL, mode);
+            if (!CheckVersion())
+                throw new IOException($"This device does not contain the correct signature 0x48 for a AK8963");
+        }
+
+        /// <summary>
+        /// Reset the device
+        /// </summary>
+        public void Reset()
+        {
+            WriteRegister(Register.RSV, 0x01);
+            // When powering the AK893, doc says 50 ms needed
+            Thread.Sleep(50);
+        }
+
+        /// <summary>
+        /// Get the device information
+        /// </summary>
+        /// <returns>The device information</returns>
+        public byte GetDeviceInfo()
+        {
+            return ReadByte(Register.INFO);
+        }
+
+        /// <summary>
+        /// Get the magnetometer bias
+        /// </summary>
+        public Vector3 MagnometerBias { get; set; } = Vector3.Zero;
+
+        /// <summary>
+        /// Calibrate the magnetometer. Make sure your sensor is as far as possible of magnet
+        /// Calculate as well the magnetometer bias
+        /// </summary>
+        /// <returns>Returns the factory calibration data</returns>
+        public Vector3 CalibrateMagnetometer()
+        {
+            Vector3 calib = new Vector3();
+            Span<byte> rawData = stackalloc byte[3];
+
+            var oldPower = MeasurementMode;
+
+            // Stop the magnetometer
+            MeasurementMode = MeasurementMode.PowerDown;
+            // Enter the magnetometer Fuse mode to read the calibration data
+            // Page 13 of documentation 
+            MeasurementMode = MeasurementMode.FuseRomAccess;
+            // Read the data
+            ReadBytes(Register.ASAX, rawData);
+            calib.X = (float)((rawData[0] - 128) / 256.0 + 1.0);
+            calib.Y = (float)((rawData[1] - 128) / 256.0 + 1.0);
+            calib.Z = (float)((rawData[2] - 128) / 256.0 + 1.0);
+            MeasurementMode = MeasurementMode.PowerDown;
+            MeasurementMode = oldPower;
+
+            // Now calculate the bias
+            // Store old mode to restore after
+            var oldMode = MeasurementMode;
+            Vector3 minbias = new Vector3();
+            Vector3 maxbias = new Vector3();
+            int numberMeasurements = 1500;
```

Comment:
should this be parameter to the function (also might be better to actually make it `measurementTime` with some sane default value)

Reply:
Calibration is by numbers of reads. Adding the time would be not the best. I can clearly make it as a parameter of the function. That's a good idea.